### PR TITLE
Snowmobiles! And something for events

### PR DESCRIPTION
**Pull request by ghost**
Adds in snowmobiles, the fastest way to get around the frozen wasteland that is our home!

also:

Memorizer: MIB inspired memory eraser for Kyres and I to use in events (it's just a resprited flash) 

Memorizer part removed pending rework.

### CODE REVIEW

File: code/game/objects/items/devices/flash.dm [hunk 1]
```diff
@@ -218,3 +218,104 @@
...
+
+			user.visible_message("<span class='disarm'> [user] shines the device into [M]'s eyes!</span>")
+		else
+
+			user.visible_message("<span class='notice'>[user] shines the device into [M]'s sensors!</span>")
+	else
+
+		user.visible_message("<span class='notice'>[user] fails to shine the device into [M]'s eyes!</span>")
+
+	return
```
Comment (Doxxmedearly) [inline code/game/objects/items/devices/flash.dm]: Are we giving any indication to the target of what this actually does to them?

**Review by Doxxmedearly — COMMENTED** [review]
Bike looks good.
Memorizer looks like a lot of repeated code? If abo's fine with that, cool. Otherwise we can tidy it; the only difference I see is in the burnt out icon (which can be moved to a var) and the flash message (Which can be moved to a proc).

File: code/game/objects/items/devices/flash.dm (hunk 1 above)
Comment (ghost) [reply code/game/objects/items/devices/flash.dm]: Wasn't sure about how to implement the proc message without repeating the code, and the plan for the future is to have a message appear saying they don't remember anything out of the ordinary that's happened.

Comment (Doxxmedearly) [reply code/game/objects/items/devices/flash.dm]: Could we do that message now?
I can help you with the proc stuff if you need!

**Comment by Doxxmedearly** [comment]
Alright. Delet the duplicate code and make changes to the flash parent's procs.
Easy part: The burnt out icon.
just do
var/burnt_icon = "burnticonstatename" in the flash parent.
so flash would be var/burnt_icon = "flashburnt" and memorizer would be burnt_icon = "memorizerburnt"
Then wherever there's icon_state = "flashburnt" replace it with icon_state = burnt_icon (Line 60, 150, 189)

For the MESSAGE
make a new proc on the flash parent. Something like /obj/item/device/flash/proc/affect_message(mob/living/M, mob/user)

then put what's around line 115 in this proc.
```
/obj/item/device/flash/proc/affect_message(mob/living/M, mob/user)
	if(!flashfail)
		flick("flash2", src)
		if(!issilicon(M))
			user.visible_message(SPAN_NOTICE("\The [user] blinds \the [M] with \the [src]!"))
		else
			user.visible_message(SPAN_NOTICE("\The [user] overloads \the [M]'s sensors with \the [src]!"))
	else
		user.visible_message(SPAN_NOTICE("\The [user] fails to blind \the [M] with \the [src]!"))
	return
```
And instead add ```affect_message(M, user)``` at line 115, deleting the stuff you put in here.

Then you can do a unique proc for the memorizer ```/obj/item/device/flash/memorizer/affect_message(mob/living/M, mob/user)```
And change the contents to whatever message you want to show

**Pull request merged**